libs/language-server/src/lib/ast/expressions/operator-type-computer.ts

```diff
@@ -0,0 +1,131 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { ValidationContext } from '../../validation/validation-context';
+import { BinaryExpression, UnaryExpression } from '../generated/ast';
+// eslint-disable-next-line import/no-cycle
+import { PropertyValuetype } from '../model-util';
+
+export interface UnaryOperatorTypeComputer {
```

Comment:
Adding a short documenting JSdoc to the interface methods might make sense to improve onboarding.